lib/vdsm/storage/lvm.py

```diff
@@ -281,6 +283,17 @@ def _buildConfig(dev_filter, use_lvmpolld="1"):
     return conf.replace("\n", " ").strip()
 
 
+def _allowed_devices(devices):
+    devs = set(d.strip() for d in chain(devices, USER_DEV_LIST))
+    devs.discard('')
+    if devs:
+        devs = sorted(d.replace(r'\x', r'\\x') for d in devs)
+        dev_list = ",".join(dev for dev in devs)
+    else:
+        dev_list = None
+    return dev_list
```

Comment:
Why do we need this? we already have code dealing with the list of devices for crating lvm filter.

We also cache the devices, and fetch the devices again when needed, but this code never consider the cached devices. Looks like you need to reuse the current code getting the devices instead of introducing a new partial helper.

Reply:
This is the code ensuring that we have correct device list:

```
 401     def _getCachedFilter(self):
 402         with self._filterLock:
 403             if self._filterStale:
 404                 self._filter = _buildFilter(multipath.getMPDevNamesIter())
 405                 self._filterStale = False
 406             return self._filter
 407 
 408     def _addExtraCfg(self, cmd, devices=tuple(), use_lvmpolld=True):
 409         newcmd = [constants.EXT_LVM, cmd[0]]
 410 
 411         if devices:
 412             dev_filter = _buildFilter(devices)
 413         else:
 414             dev_filter = self._getCachedFilter()
```

This code should be refactored so we cache the devices, not the filter, and we use
the cached device when crating a filter or formatting the --devices option.

Also when invalidating the cache, the next command should get new list of devices, same
way it happens now with the filter.

This refactoring must come first, before we add devices support, since devices support depends on it.